Hot Fix: `--config` is being ignored

archerzz opened the pull request:
- fix an expression typo when determining the value of `configPath`
- add test cases

fix #2277

timotheeguerin commented:
Should we do a hot fix for this?

timotheeguerin commented on packages/compiler/test/config/config.test.ts:
```diff
@@ -1,19 +1,25 @@
-import { deepStrictEqual, strictEqual } from "assert";
-import { dirname, join, resolve } from "path";
+import { deepStrictEqual, fail, strictEqual } from "assert";
+import { dirname, join, normalize, resolve } from "path";
```
Should try not to use anything from `path` anywhere but scripts. This cause issues with windows vs unix path and everything in the compiler should already have been normalized to unix paths. The compiler provide some path functions that can be used instead.

timotheeguerin commented on packages/compiler/test/config/config.test.ts:
```diff
@@ -148,3 +154,58 @@ describe("compiler: config file loading", () => {
+
+    it("resolves full path to custom config file", async () => {
+      const config = await resolveOptions("custom/myConfig.yaml");
+      config.options.config = normalize(config.options.config!);
```
Like having to do that just so you can compare on both platform

timotheeguerin commented on packages/compiler/test/config/config.test.ts:
```diff
@@ -148,3 +154,58 @@ describe("compiler: config file loading", () => {
+      fail("should throw error");
+    });
+
+    it("resolves yaml config file", async () => {
```
A bit confused by what this test is testing, shouldn't `--config` always only point to a config file? Do we allow it to be a directory too?

timotheeguerin replied:
Filed an issue to discuss this behavior https://github.com/microsoft/typespec/issues/2282. Please comment on it if it was something you expected to work and depended on.

timotheeguerin commented on common/changes/@typespec/compiler/fix-config-option_2023-08-10-08-29.json:
```diff
@@ -0,0 +1,10 @@
+  "changes": [
+    {
+      "packageName": "@typespec/compiler",
+      "comment": "fix(config): --config is not adopted",
```
This shows up in the changelog, it should be a bit more descriptive too

timotheeguerin commented:
Thanks for the Fix @archerzz I converted this to be an hotfix or I guess you will be prevented from migrating to 0.47. Also fixed up the comments I made on the PR and filed an issue about a behavior you added a test for which I feel is unexpected https://github.com/microsoft/typespec/issues/2282(Please comment on it if you are depending on it)

archerzz commented:
> Thanks for the Fix @archerzz I converted this to be an hotfix or I guess you will be prevented from migrating to 0.47. Also fixed up the comments I made on the PR and filed an issue about a behavior you added a test for which I feel is unexpected #2282 comment on it if you are depending on it)

Thanks!
